catalog/transformer/catalog-transformer-preview/src/main/java/org/codice/ddf/transformer/preview/PreviewMetacardTransformer.java

```diff
@@ -34,15 +57,62 @@ public BinaryContent transform(Metacard metacard, Map<String, Serializable> argu
     }
 
     String preview = "No preview text available.";
+
     if (metacard.getAttribute(Extracted.EXTRACTED_TEXT) != null
         && metacard.getAttribute(Extracted.EXTRACTED_TEXT).getValue() != null) {
       preview =
           StringEscapeUtils.escapeHtml4(
                   metacard.getAttribute(Extracted.EXTRACTED_TEXT).getValue().toString())
               .replaceAll("[\n|\r]", "<br>");
+    } else if (previewFromMetadata && StringUtils.isNotEmpty(metacard.getMetadata())) {
+      preview =
+          StringEscapeUtils.escapeHtml4(selectPreviewFromMetadata(metacard.getMetadata()))
+              .replaceAll("[\n|\r]", "<br>");
+    }
+
+    if (StringUtils.isNotEmpty(preview)) {
       preview = String.format("<head><meta charset=\"utf-8\"/>%s</head>", preview);
     }
 
     return new BinaryContentImpl(IOUtils.toInputStream(preview));
   }
+
+  private String selectPreviewFromMetadata(String metadata) {
+    XPathFactory xPathFactory;
+    String xPathString = "//*[name()='%s']";
+    String text = "";
+
+    for (String attributeName : textElements) {
```

Comment:
✏️ You could use a Java8 stream instead of this for loop. With my other comment, it'd look like
```
    textXpaths.stream()
        .map(x -> {
          StringReader metadataReader = new StringReader(metadata);
          InputSource inputXml = new InputSource(metadataReader);
          return x.evaluate(inputXml, XPathConstants.NODE);
        })
        .filter(Objects::nonNull)
        .findFirst();
```
You'll probably have to surround this whole streaming logic with the try-catch.